src/main/java/core/basesyntax/strategy/OperationHandler.java

```diff
@@ -0,0 +1,5 @@
+package core.basesyntax.strategy;
+
+public interface OperationHandler {
+    int executeOperation(int currentQuantity, int quantityFromTransaction);
```

Comment:
Overall such approach is ok, I cannot say that this is mistake. 
But for current task and in scope of further testing I'd recommend to rework handlers to put new result directly to storage (and obtain current quantity from there), not just calculate result.
With such approach you will be able to remove FruitsDao totally and make logic more simple in TransactionHandlerImpl.

```suggestion
    void executeOperation(int quantityFromTransaction);
```